### DISCUSSION TITLE
How to use resend.com api for sending emails?

### DISCUSSION THREAD

**Discussion by Demoevery** [pocketbase/pocketbase#2966] [Q&A] [answered]
Resend allows to send email using their api. What would be the way to send email in this manner in pocketbase?
```
package main

import "github.com/resendlabs/resend-go"

func main() {
    apiKey = "re_123"

    client := resend.NewClient(apiKey)

    params := &resend.SendEmailRequest{
        From:    "Acme <onboarding@resend.dev>",
        To:      []string{"delivered@resend.dev"},
        Html:    "<strong>hello world</strong>",
        Subject: "Hello from Golang",
        Cc:      []string{"cc@example.com"},
        Bcc:     []string{"bcc@example.com"},
        ReplyTo: "replyto@example.com",
    }

    sent, err := client.Emails.Send(params)
    if err != nil {
        fmt.Println(err.Error())
        return
    }
    fmt.Println(sent.Id)
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
In order to use the builtin PocketBase email client (see "[Sending custom emails](https://pocketbase.io/docs/sending-emails/#send-custom-email)") you need a **provider that supports SMTP**.

Looking at [Resend API documentation](https://resend.com/docs/api-reference/introduction) there is no mention for SMTP so I'm not sure if it is supported.

In general you can initialize their SDK client as shown in your example. To intercept the system emails action (eg. for password reset) you can attach to the specific [`OnMailer*` hook](https://pocketbase.io/docs/event-hooks/#mailer-hooks) and return `hook.StopPropagation`. Something like:

```go
package main

import (
    "log"

    "github.com/pocketbase/pocketbase"
    "github.com/pocketbase/pocketbase/core"
    "github.com/pocketbase/pocketbase/tools/hook"
)

func main() {
    app := pocketbase.New()

    app.OnMailerBeforeRecordResetPasswordSend().Add(func(e *core.MailerRecordEvent) error {
        log.Println(e.Message.Subject)
        log.Println(e.Message.From)
        log.Println(e.Message.To)
        log.Println(e.Message.HTML)

        // use the above fields to send the email with the custom Resend SDK client...

        return hook.StopPropagation
    })

    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

In case Resend is not your final choice, here are some mail providers that have free tier and support SMTP:
- [Mailjet](https://www.mailjet.com/pricing/)
- [Sendinblue/Brevo](https://www.brevo.com/pricing/)
- [SendGrid](https://sendgrid.com/pricing/)
- ~[Mailgun](https://www.mailgun.com/)~ _it seems that they've switched to only 1 month trial_
- _and probably others..._

_[Mailtrap](https://mailtrap.io/email-sending/) also recently added, in addition to Email testing, support for [Email sending](https://mailtrap.io/email-sending/) with pretty generous free plan (1k mails/month without credit card) but I haven't personally tested them yet._

**Reply by Demoevery** [reply]
Thank you for your quick response. That seems to be what I needed.
What I like about resend is that they don't require smtp server setup. Maybe a feature could be added so that anyone can use resend just by adding api key. Supabase also uses for it's email implementation I think.

**Reply by ganigeorgiev** [reply]
> What I like about resend is that they don't require smtp server setup. 

You don't have to setup your own SMTP server. You usually have to specify only the mail host address, username and password (see  https://pocketbase.io/docs/going-to-production/#use-smtp-mail-server).

> Maybe a feature could be added so that anyone can use resend just by adding api key. Supabase also uses for it's email implementation I think.

Sorry, there are no plans to add support for vendor specific APIs as it will be hard to maintain. SMTP is standard protocol, old almost as the internet, and it should be supported by most email providers already. Resend seems to be more of an exception as it is still fairly new and I think they just haven't got the time to work on it yet.
